### ISSUE TITLE
Some bugs about value bounds

### ISSUE THREAD

**Issue by kquine** [dreal/dreal3#114]
The following example is satisfiable for `k=0`.

```
#define g           9.80555
#define pi          3.1415926535897931
#define m           20500.0
#define qbw         (13.97 * 30 * 300)
#define qw          (13.97 * 300)
#define V           111.64
#define c1          -0.770
#define c2          0.02755
#define c3          1.055E-4
#define c4          1.642E-6
#define c8          -0.7336
#define c9          1.587E-5

#define cy(a,r,b)   (qw * (-0.02 * 180 / pi * b + 0.021 * a / 20 + 0.086 * r / 30))
#define cl(a,r,b)   (qbw * (-0.0008 * 180 / pi * b + 0.05 * a / 20 + 0.013 * r / 30))
#define cn(a,r,b)   (qbw * (0.02 * 180 / pi * b + -0.01 * a / 20 + -0.04 * r / 30))

[-1.5,1.5]    beta;
[-1.5,1.5]    p;
[-1.5,1.5]    r;
[-1.5,1.5]    phi;
[-1.5,1.5]    psi;

[-60,60]    xAIL;
[-60,60]    xRDR;
[-60,60]    rAIL;
[-60,60]    rRDR;
[-60,60]    gAIL;
[-60,60]    gRDR;
[-60,60]    gDir;

[0,0.5]  tau;
[0,0.5]  time;

{mode 1;
  invt: (tau >= 0);
        (tau <= 0.5);
  flow:
        d/dt[beta]  = cy(xAIL,xRDR,beta) / (m * V) - r + (V / g) * cos(beta) * sin(phi);
        d/dt[phi]   = p;
        d/dt[psi]   = (g / V) * tan(phi);
        d/dt[p]     = (c1 * r + c2 * p) * r * tan(phi) + c3 * cl(xAIL,xRDR,beta)
 + c4 * cn(xAIL,xRDR,beta);
        d/dt[r]     = (c8 * p - c2 * r) * r * tan(phi) + c4 * cl(xAIL,xRDR,beta)
 + c9 * cn(xAIL,xRDR,beta);
        d/dt[xAIL]  = rAIL;
        d/dt[xRDR]  = rRDR;
        d/dt[rAIL]  = 0;
        d/dt[rRDR]  = 0;
        d/dt[gAIL]  = 0;
        d/dt[gRDR]  = 0;
        d/dt[gDir]  = 0;
        d/dt[tau]   = 1;
  jump:
        (and (tau = 0.5)) ==>
        @1 (and (beta' = beta) (phi' = phi) (psi' = psi) (p' = p) (r' = r)
                (xAIL' = xAIL) (xRDR' = xRDR)
                (rAIL' = (gAIL' - xAIL) /0.5) (rRDR' = (gRDR' - xRDR) /0.5)
                (gAIL' = (gDir - psi*180/3.14) * 0.5) (gRDR' = beta * 0.35) (gDir' = gDir) (tau' = 0));
}

init: @1 (and (beta = 0)
              (phi = 0)
              (psi = 0)
              (p = 0)
              (r = 0)
              (xAIL = 0)
              (xRDR = 0)
              (rAIL = 0)
              (rRDR = 0)
              (gAIL = 0)
              (gRDR = 0)
              (tau = 0));
goal: @1 true;
```

However, if I remove `(phi = 0)` from the initial condition (i.e., allowing a wider bound for `phi`), then it becomes `unsat`, or terminates without output (the generated `output` file is empty).

```
#define g           9.80555
#define pi          3.1415926535897931
#define m           20500.0
#define qbw         (13.97 * 30 * 300)
#define qw          (13.97 * 300)
#define V           111.64
#define c1          -0.770
#define c2          0.02755
#define c3          1.055E-4
#define c4          1.642E-6
#define c8          -0.7336
#define c9          1.587E-5

#define cy(a,r,b)   (qw * (-0.02 * 180 / pi * b + 0.021 * a / 20 + 0.086 * r / 30))
#define cl(a,r,b)   (qbw * (-0.0008 * 180 / pi * b + 0.05 * a / 20 + 0.013 * r / 30))
#define cn(a,r,b)   (qbw * (0.02 * 180 / pi * b + -0.01 * a / 20 + -0.04 * r / 30))

[-1.5,1.5]    beta;
[-1.5,1.5]    p;
[-1.5,1.5]    r;
[-1.5,1.5]    phi;
[-1.5,1.5]    psi;

[-60,60]    xAIL;
[-60,60]    xRDR;
[-60,60]    rAIL;
[-60,60]    rRDR;
[-60,60]    gAIL;
[-60,60]    gRDR;
[-60,60]    gDir;

[0,0.5]  tau;
[0,0.5]  time;

{mode 1;
  invt: (tau >= 0);
        (tau <= 0.5);
  flow:
        d/dt[beta]  = cy(xAIL,xRDR,beta) / (m * V) - r + (V / g) * cos(beta) * sin(phi);
        d/dt[phi]   = p;
        d/dt[psi]   = (g / V) * tan(phi);
        d/dt[p]     = (c1 * r + c2 * p) * r * tan(phi) + c3 * cl(xAIL,xRDR,beta)
 + c4 * cn(xAIL,xRDR,beta);
        d/dt[r]     = (c8 * p - c2 * r) * r * tan(phi) + c4 * cl(xAIL,xRDR,beta)
 + c9 * cn(xAIL,xRDR,beta);
        d/dt[xAIL]  = rAIL;
        d/dt[xRDR]  = rRDR;
        d/dt[rAIL]  = 0;
        d/dt[rRDR]  = 0;
        d/dt[gAIL]  = 0;
        d/dt[gRDR]  = 0;
        d/dt[gDir]  = 0;
        d/dt[tau]   = 1;
  jump:
        (and (tau = 0.5)) ==>
        @1 (and (beta' = beta) (phi' = phi) (psi' = psi) (p' = p) (r' = r)
                (xAIL' = xAIL) (xRDR' = xRDR)
                (rAIL' = (gAIL' - xAIL) /0.5) (rRDR' = (gRDR' - xRDR) /0.5)
                (gAIL' = (gDir - psi*180/3.14) * 0.5) (gRDR' = beta * 0.35) (gDir' = gDir) (tau' = 0));
}

init: @1 (and (beta = 0)
              (psi = 0)
              (p = 0)
              (r = 0)
              (xAIL = 0)
              (xRDR = 0)
              (rAIL = 0)
              (rRDR = 0)
              (gAIL = 0)
              (gRDR = 0)
              (tau = 0));
goal: @1 true;
```

**Comment by soonhokong** [comment]
We have a segmentation fault for the second example, and dReach interprets it as an `UNSAT`. I'll fix the segmentation fault first, and fix dReach to handle such an exception from dReal. I've added the example as [src/tests/nra_ode/kquine_04.smt2](https://github.com/dreal/dreal3/blob/1cbb1e1c933f43eab75a837903a85a9a2b00fca6/src/tests/nra_ode/kquine_04.smt2).

**Comment by soonhokong** [comment]
@kquine, I fixed the bug in 9196b87. I also had to modify src/tests/kquine_04.smt2 file to narrow the domain of `phi_0_0` variable to `[-0.005, 0.005]` to make it easier to test it (otherwise, it seems it takes more than 5min).
